`yield` fails Indentation check in switch expressions

I have read check documentation: https://checkstyle.sourceforge.io/checks/misc/indentation.html
I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words


```bash
$ javac IndentationFoo.java
$

```

```bash
$ cat checkstyle-foo.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="Indentation">
            <property name="forceStrictCondition" value="true"/>
        </module>
    </module>
</module>
```

```bash
$ cat IndentationFoo.java 
import java.time.DayOfWeek;
import java.time.LocalDate;

public class IndentationFoo {
    public static void main(final String[] args) {
        final DayOfWeek today = LocalDate.now().getDayOfWeek();
        final boolean isWeekend = switch (today) {
            case SATURDAY:
            System.out.println("Saturday"); // Warning: expected indentation: 16
            yield true; // Warning: expected indentation: 16.
            case SUNDAY:
                System.out.println("Sunday"); // (no warning)
                yield true; // Warning: expected indentation: 12
            default:
yield false; // Warning: expected indentation: 12
        };
        System.out.println("isWeekend = " + isWeekend);
    }
}
```

```bash
$ java -jar checkstyle-10.20.1-all.jar -c checkstyle-foo.xml IndentationFoo.java 
Starting audit...
[ERROR] IndentationFoo.java:9:13: 'block' child has incorrect indentation level 12, expected level should be 16. [Indentation]
[ERROR] IndentationFoo.java:10:13: 'yield' has incorrect indentation level 12, expected level should be 16. [Indentation]
[ERROR] IndentationFoo.java:13:17: 'yield' has incorrect indentation level 16, expected level should be 12. [Indentation]
[ERROR] IndentationFoo.java:15:1: 'yield' has incorrect indentation level 0, expected level should be 12. [Indentation]
Audit done.
Checkstyle ends with 4 errors.

```
---

The Indentation check rejects all `yield` statements in switch expressions when `forceStrictCondition` is set to true. In the example above, it rejects anything that is not 12, saying 12 is the correct level. If you set the indentation level to 12, it rejects it saying it should be 16. 

This is also inconsistent with the required indentation level for other statements, which require an indentation of 16. 

I would expect 16 to be accepted as the correct indentation for all of these statements.

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/LineWrappingHandler.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/LineWrappingHandler.java
@@ -84,6 +84,7 @@ public class LineWrappingHandler {
         TokenTypes.LCURLY,
         TokenTypes.RCURLY,
         TokenTypes.LITERAL_NEW,
+        TokenTypes.LITERAL_YIELD,
         TokenTypes.ARRAY_INIT,
         TokenTypes.LITERAL_DEFAULT,
         TokenTypes.LITERAL_CASE,

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java
@@ -1351,6 +1351,25 @@ public class IndentationCheckTest extends AbstractModuleTestSupport {
     }
 
     @Test
+    public void testYieldKeywordWithForceStrictCondition() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
+        checkConfig.addProperty("forceStrictCondition", "true");
+        checkConfig.addProperty("tabWidth", "4");
+        final String[] expected = {
+            "15:13: " + getCheckMessage(MSG_CHILD_ERROR, "block", 12, 16),
+            "16:13: " + getCheckMessage(MSG_ERROR, "yield", 12, 16),
+            "44:13: " + getCheckMessage(MSG_CHILD_ERROR, "block", 12, 16),
+            "45:13: " + getCheckMessage(MSG_ERROR, "yield", 12, 16),
+            "50:5: " + getCheckMessage(MSG_ERROR, "yield", 4, 16),
+            "71:15: " + getCheckMessage(MSG_ERROR, "yield", 14, 16),
+            "74:20: " + getCheckMessage(MSG_ERROR, "yield", 19, 16),
+            "77:9: " + getCheckMessage(MSG_ERROR, "yield", 8, 16),
+        };
+        verifyWarns(checkConfig,
+                getNonCompilablePath("InputIndentationYieldForceStrict.java"), expected);
+    }
+
+    @Test
     public void testChainedMethodCalling() throws Exception {
         final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
 

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationYieldForceStrict.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationYieldForceStrict.java
@@ -1,0 +1,98 @@
+//non-compiled with javac: Compilable with Java17                        //indent:0 exp:0
+
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation;  //indent:0 exp:0
+
+/* Config:                                                               //indent:0 exp:0
+ *                                                                       //indent:1 exp:1
+ * forceStrictCondition = true                                           //indent:1 exp:1
+ */                                                                      //indent:1 exp:1
+
+class InputIndentationYieldForceStrict {                                 //indent:0 exp:0
+    public static void main(final String[] args) {                       //indent:4 exp:4
+        final int today = 0;                                             //indent:8 exp:8
+        final boolean isWeekDay = switch (today) {                       //indent:8 exp:8
+            case 2 -> {                                                  //indent:12 exp:12
+            System.out.println("Monday");                                //indent:12 exp:16 warn
+            yield true;                                                  //indent:12 exp:16 warn
+            }                                                            //indent:12 exp:12
+            case 3 -> {                                                  //indent:12 exp:12
+                System.out.println("Tuesday");                           //indent:16 exp:16
+                yield true;                                              //indent:16 exp:16
+            }                                                            //indent:12 exp:12
+            default -> {                                                 //indent:12 exp:12
+                yield true;                                              //indent:16 exp:16
+            }                                                            //indent:12 exp:12
+        };                                                               //indent:8 exp:8
+
+        final int tomorrow = 1;                                          //indent:8 exp:8
+        final boolean isTomorrowWeekDay = switch (tomorrow) {            //indent:8 exp:8
+            case 3 -> {                                                  //indent:12 exp:12
+                System.out.println("Monday");                            //indent:16 exp:16
+                yield true;                                              //indent:16 exp:16
+            }                                                            //indent:12 exp:12
+            case 4 -> {                                                  //indent:12 exp:12
+                System.out.println("Tuesday");                           //indent:16 exp:16
+                yield true;                                              //indent:16 exp:16
+            }                                                            //indent:12 exp:12
+            default -> {                                                 //indent:12 exp:12
+                yield true;                                              //indent:16 exp:16
+            }                                                            //indent:12 exp:12
+        };                                                               //indent:8 exp:8
+
+        final boolean isWeekend = switch (today) {                       //indent:8 exp:8
+            case 0:                                                      //indent:12 exp:12
+            System.out.println("Saturday");                              //indent:12 exp:16 warn
+            yield true;                                                  //indent:12 exp:16 warn
+            case 1:                                                      //indent:12 exp:12
+                System.out.println("Sunday");                            //indent:16 exp:16
+                yield true;                                              //indent:16 exp:16
+            default:                                                     //indent:12 exp:12
+    yield true;                                                          //indent:4 exp:16 warn
+        };                                                               //indent:8 exp:8
+
+        final boolean isWeekendTom = switch (tomorrow) {                 //indent:8 exp:8
+            case 1:                                                      //indent:12 exp:12
+                System.out.println("Saturday");                          //indent:16 exp:16
+                yield true;                                              //indent:16 exp:16
+            case 2:                                                      //indent:12 exp:12
+                System.out.println("Sunday");                            //indent:16 exp:16
+                yield true;                                              //indent:16 exp:16
+            default:                                                     //indent:12 exp:12
+                yield true;                                              //indent:16 exp:16
+        };                                                               //indent:8 exp:8
+    }                                                                    //indent:4 exp:4
+
+    public boolean returnKeywordWrong(int k) {                           //indent:4 exp:4
+        return switch (k) {                                              //indent:8 exp:8
+            case 1 -> {                                                  //indent:12 exp:12
+                yield false;                                             //indent:16 exp:16
+            }                                                            //indent:12 exp:12
+            case 2 -> {                                                  //indent:12 exp:12
+              yield true;                                                //indent:14 exp:16 warn
+            }                                                            //indent:12 exp:12
+            case 3 -> {                                                  //indent:12 exp:12
+                   yield true;                                           //indent:19 exp:16 warn
+            }                                                            //indent:12 exp:12
+            default -> {                                                 //indent:12 exp:12
+        yield false;                                                     //indent:8 exp:16 warn
+            }                                                            //indent:12 exp:12
+        };                                                               //indent:8 exp:8
+    }                                                                    //indent:4 exp:4
+
+    public boolean returnKeywordCorrect(int k) {                         //indent:4 exp:4
+        return switch (k) {                                              //indent:8 exp:8
+            case 1 -> {                                                  //indent:12 exp:12
+                yield false;                                             //indent:16 exp:16
+            }                                                            //indent:12 exp:12
+            case 2 -> {                                                  //indent:12 exp:12
+                yield true;                                              //indent:16 exp:16
+            }                                                            //indent:12 exp:12
+            case 3 -> {                                                  //indent:12 exp:12
+                yield true;                                              //indent:16 exp:16
+            }                                                            //indent:12 exp:12
+            default -> {                                                 //indent:12 exp:12
+                yield false;                                             //indent:16 exp:16
+            }                                                            //indent:12 exp:12
+        };                                                               //indent:8 exp:8
+    }                                                                    //indent:4 exp:4
+}                                                                        //indent:0 exp:0

